src/HealthChecks.Aws.S3/S3HealthCheck.cs

```diff
@@ -1,53 +1,69 @@
-using Amazon.S3;
-using Microsoft.Extensions.Diagnostics.HealthChecks;
-
-namespace HealthChecks.Aws.S3
-{
-    public class S3HealthCheck : IHealthCheck
-    {
-        private readonly S3BucketOptions _bucketOptions;
-
-        public S3HealthCheck(S3BucketOptions bucketOptions)
-        {
+using Amazon.Runtime;
+using Amazon.S3;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.Aws.S3
+{
+    public class S3HealthCheck : IHealthCheck
+    {
+        private readonly S3BucketOptions _bucketOptions;
+
+        public S3HealthCheck(S3BucketOptions bucketOptions)
+        {
             if (bucketOptions == null)
             {
                 throw new ArgumentNullException(nameof(bucketOptions));
-            }
+            }
+
             if (bucketOptions.S3Config == null)
             {
                 throw new ArgumentNullException(nameof(S3BucketOptions.S3Config));
-            }
-            _bucketOptions = bucketOptions;
-        }
-
-        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
-        {
-            try
-            {
-                bool keysProvided = !string.IsNullOrEmpty(_bucketOptions.AccessKey) &&
-                                    !string.IsNullOrEmpty(_bucketOptions.SecretKey);
-
-                var client = keysProvided
-                    ? new AmazonS3Client(_bucketOptions.AccessKey, _bucketOptions.SecretKey, _bucketOptions.S3Config)
-                    : new AmazonS3Client(_bucketOptions.S3Config);
+            }
+            _bucketOptions = bucketOptions;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                AWSCredentials credentials = _bucketOptions.Credentials;
+                if (credentials == null)
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    if (!string.IsNullOrEmpty(_bucketOptions.AccessKey) && !string.IsNullOrEmpty(_bucketOptions.SecretKey))
+                    {
+                        // for backwards compatibility we create the basic credentials if the old fields are used
+                        // but if they are not specified we fallback to using the default profile
+                        credentials = new BasicAWSCredentials(_bucketOptions.AccessKey, _bucketOptions.SecretKey);
+                    }
+#pragma warning restore CS0618 // Type or member is obsolete
+                }
+
+                bool keysProvided = credentials != null;
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var client = keysProvided
+                    ? new AmazonS3Client(_bucketOptions.AccessKey, _bucketOptions.SecretKey, _bucketOptions.S3Config)
+                    : new AmazonS3Client(credentials, _bucketOptions.S3Config);

```

Comment:
I don't understand your changes. You use `credentials` variable if it is `null`! My suggestion was about first branch of ternary operator.

Reply:
I see your confusion, this code changed quite a bit from my original change and I messed it up when rebasing, didn't realize it was using S3Client without any credentials specified. Updated this to use the credentials if they have been specified in any way, if not falling back to S3Client default behavior (as we did before)